test/jdk/java/awt/dnd/DragSourceMotionListenerTest.java

```diff
@@ -0,0 +1,245 @@
+    private static final Point testPoint1 = new Point();
+    private static final Point testPoint2 = new Point();
+    private static volatile Point srcPoint;
+    private static volatile Point dstOutsidePoint;
+    private static volatile Point dstInsidePoint;
+
+    private static final Transferable t = new StringSelection("TEXT");
+    private static final DragGestureListener gestureListener = e -> e.startDrag(null, t);
+
+    private static final DragSourceAdapter sourceAdapter = new DragSourceAdapter() {
+        public void dragMouseMoved(DragSourceDragEvent dsde) {
+            if (Math.abs(dsde.getX() - testPoint1.getX()) < 5) {
+                passedTest1 = true;
+            }
+        }
+
+        public void dragDropEnd(DragSourceDropEvent dsde) {
+            if (Math.abs(dsde.getX() - testPoint2.getX()) < 5) {
+                passedTest2 = true;
+            }
+        }
+    };
+
+    private static final DropTargetListener targetAdapter = new DropTargetAdapter() {
+        public void drop(DropTargetDropEvent e) {
+            e.acceptDrop(DnDConstants.ACTION_COPY);
+            try {
+                final Transferable t = e.getTransferable();
+                final String str =
+                        (String) t.getTransferData(DataFlavor.stringFlavor);
+                e.dropComplete(true);
+            } catch (Exception ex) {
+                ex.printStackTrace();
+                e.dropComplete(false);
+            }
+        }
+    };
+
+    private static final DropTarget dropTarget = new DropTarget(target, targetAdapter);
+    Component clickedComponent = null;
+
+    private void createAndShowUI() {
+        frame = new Frame("DragSourceMotionListenerTest");
+        ds.addDragSourceListener(sourceAdapter);
+        ds.addDragSourceMotionListener(sourceAdapter);
+        ds.createDefaultDragGestureRecognizer(source, DnDConstants.ACTION_COPY, gestureListener);
+        target.setDropTarget(dropTarget);
+
+        frame.setLayout(new GridLayout(1, 2));
+
+        frame.add(source);
+        frame.add(target);
+
+        Toolkit.getDefaultToolkit()
+               .addAWTEventListener(this, AWTEvent.MOUSE_EVENT_MASK);
+        frame.pack();
+        frame.setVisible(true);
+    }
+
+    public static void main(String[] args) throws Exception {
+        try {
+            Robot robot = new Robot();
+            robot.setAutoDelay(10);
+
+            DragSourceMotionListenerTest dsmObj = new DragSourceMotionListenerTest();
+            EventQueue.invokeAndWait(dsmObj::createAndShowUI);
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            EventQueue.invokeAndWait(() -> {
+                Point p = getPoint(source, 1);
+                srcPoint = p;
+
+                p = getPoint(frame, 3);
+                dstOutsidePoint = p;
+                testPoint1.setLocation(p);
+
+                p = getPoint(target, 1);
+                dstInsidePoint = p;
+                testPoint2.setLocation(p);
```

Comment:
Now the local variable isn't needed any more, all the magic happens inside `getPoint`:

```suggestion
                srcPoint = getPoint(source, 1);

                dstOutsidePoint = getPoint(frame, 3);
                testPoint1.setLocation(dstOutsidePoint);

                dstInsidePoint = getPoint(target, 1);
                testPoint2.setLocation(dstInsidePoint);
```

In fact, I meant getting rid of the lambda and moving its body into the helper method with the local variables as you wrote it and as I suggested to modify.

At the same time, this way also works: you moved out the common logic, and the lambda expression is not as long.